src/lang/internals/sclang-io.js

```diff
@@ -15,9 +15,12 @@ export const STATES = {
   COMPILED: 'compiled',
   COMPILING: 'compiling',
   COMPILE_ERROR: 'compileError',
-  READY: 'ready'
+  READY: 'ready',
+  CAPTURING: 'capturing'
```

Comment:
It adds a new STATES "CAPTURING", but the intention of the states is only to track the major lifecycle of the sclang interpreter. As you have it now it would go back and forth from READY to CAPTURING.

I think you are assuming that there can only be one interpreter call active at a time. In fact, you can launch many of them and each one many do async things over in supercollider lang, and return results or errors at any point in the future.